### ISSUE TITLE
Twitter spaces HTTP Error 403: Forbidden

### ISSUE THREAD

**Issue by willmcginnis** [yt-dlp/yt-dlp#8029]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I'm running yt-dlp version **2023.07.06** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [ ] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

California United States

### Provide a description that is worded well enough to be understood

Twitter does require that you login to play but otherwise the audio is publicly available in that any account should be able to listen. I searched for about a half hour but didn't find any other bugs related to this but apologies if I missed somethinga bout it.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
$ yt-dlp --cookies ~/cookies\ twitter.txt -vU "https://twitter.com/i/spaces/1ypKddqewndKW?s=20"
[debug] Command-line config: ['--cookies', '/home/will/cookies twitter.txt', '-vU', 'https://twitter.com/i/spaces/1ypKddqewndKW?s=20']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (zip)
[debug] Python 3.9.2 (CPython x86_64 64bit) - Linux-5.10.0-25-amd64-x86_64-with-glibc2.31 (OpenSSL 1.1.1n  15 Mar 2022, glibc 2.31)
[debug] exe versions: ffmpeg 4.3.6-0, ffprobe 4.3.6-0
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
Current Build Hash: cedf44a9cf3c241520a692c1006d276e6f434d4d945727c47cb1a70d32953f4a
yt-dlp is up to date (stable@2023.07.06)
[twitter:spaces] Extracting URL: https://twitter.com/i/spaces/1ypKddqewndKW?s=20
[twitter:spaces] 1ypKddqewndKW: Downloading GraphQL JSON
[twitter:spaces] 28_1698825972034961408: Downloading legacy API JSON
[twitter:spaces] 28_1698825972034961408: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 1ypKddqewndKW: Downloading 1 format(s): 0
[debug] Invoking hlsnative downloader on "https://prod-fastly-ap-south-1.video.pscp.tv/Transcoding/v1/hls/UL9-C36LhISy6bfToPyaGokh3Xya9JZhSUocgOCtw7BduTNobFWiYZMsiiXp9JYQZij7suGAzIwtfU7h6BNgCw/non_transcode/ap-southeast-1/periscope-replay-direct-prod-ap-southeast-1-public/audio-space/playlist_16752870527289245639.m3u8?type=replay"
[hlsnative] Downloading m3u8 manifest
ERROR: unable to download video data: HTTP Error 403: Forbidden
Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3363, in process_info
    success, real_download = self.dl(temp_filename, info_dict)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3084, in dl
    return fd.download(name, new_info, subtitle)
  File "/usr/local/bin/yt-dlp/yt_dlp/downloader/common.py", line 461, in download
    ret = self.real_download(filename, info_dict)
  File "/usr/local/bin/yt-dlp/yt_dlp/downloader/hls.py", line 77, in real_download
    urlh = self.ydl.urlopen(self._prepare_url(info_dict, man_url))
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3863, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/usr/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.9/urllib/request.py", line 561, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 641, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

**Comment by willmcginnis** [comment]
Likely relevant is that it's trying to get video by default but it's audio only:

```
$ yt-dlp --cookies ~/cookies\ twitter.txt -vUF "https://twitter.com/i/spaces/1ypKddqewndKW?s=20"
[debug] Command-line config: ['--cookies', '/home/will/cookies twitter.txt', '-vUF', 'https://twitter.com/i/spaces/1ypKddqewndKW?s=20']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (zip)
[debug] Python 3.9.2 (CPython x86_64 64bit) - Linux-5.10.0-25-amd64-x86_64-with-glibc2.31 (OpenSSL 1.1.1n  15 Mar 2022, glibc 2.31)
[debug] exe versions: ffmpeg 4.3.6-0, ffprobe 4.3.6-0
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
Current Build Hash: cedf44a9cf3c241520a692c1006d276e6f434d4d945727c47cb1a70d32953f4a
yt-dlp is up to date (stable@2023.07.06)
[twitter:spaces] Extracting URL: https://twitter.com/i/spaces/1ypKddqewndKW?s=20
[twitter:spaces] 1ypKddqewndKW: Downloading GraphQL JSON
[twitter:spaces] 28_1698825972034961408: Downloading legacy API JSON
[twitter:spaces] 28_1698825972034961408: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] Available formats for 1ypKddqewndKW:
ID EXT RESOLUTION │ PROTO │ VCODEC     ACODEC MORE INFO
───────────────────────────────────────────────────────
0  m4a audio only │ m3u8  │ audio only aac    m4a_dash```

**Comment by willmcginnis** [comment]
But trying to specify the one available format ID doesn't change the error:
```
$ yt-dlp -v -f 0 --cookies ~/cookies\ twitter.txt -vU "https://twitter.com/i/spaces/1ypKddqewndKW?s=20"
[debug] Command-line config: ['-v', '-f', '0', '--cookies', '/home/will/cookies twitter.txt', '-vU', 'https://twitter.com/i/spaces/1ypKddqewndKW?s=20']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version stable@2023.07.06 [b532a3481] (zip)
[debug] Python 3.9.2 (CPython x86_64 64bit) - Linux-5.10.0-25-amd64-x86_64-with-glibc2.31 (OpenSSL 1.1.1n  15 Mar 2022, glibc 2.31)
[debug] exe versions: ffmpeg 4.3.6-0, ffprobe 4.3.6-0
[debug] Optional libraries: sqlite3-2.6.0
[debug] Proxy map: {}
[debug] Loaded 1855 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releases/latest
Available version: stable@2023.07.06, Current version: stable@2023.07.06
Current Build Hash: cedf44a9cf3c241520a692c1006d276e6f434d4d945727c47cb1a70d32953f4a
yt-dlp is up to date (stable@2023.07.06)
[twitter:spaces] Extracting URL: https://twitter.com/i/spaces/1ypKddqewndKW?s=20
[twitter:spaces] 1ypKddqewndKW: Downloading GraphQL JSON
[twitter:spaces] 28_1698825972034961408: Downloading legacy API JSON
[twitter:spaces] 28_1698825972034961408: Downloading m3u8 information
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[info] 1ypKddqewndKW: Downloading 1 format(s): 0
[debug] Invoking hlsnative downloader on "https://prod-fastly-ap-south-1.video.pscp.tv/Transcoding/v1/hls/UL9-C36LhISy6bfToPyaGokh3Xya9JZhSUocgOCtw7BduTNobFWiYZMsiiXp9JYQZij7suGAzIwtfU7h6BNgCw/non_transcode/ap-southeast-1/periscope-replay-direct-prod-ap-southeast-1-public/audio-space/playlist_16752870527289245639.m3u8?type=replay"
[hlsnative] Downloading m3u8 manifest
ERROR: unable to download video data: HTTP Error 403: Forbidden
Traceback (most recent call last):
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3363, in process_info
    success, real_download = self.dl(temp_filename, info_dict)
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3084, in dl
    return fd.download(name, new_info, subtitle)
  File "/usr/local/bin/yt-dlp/yt_dlp/downloader/common.py", line 461, in download
    ret = self.real_download(filename, info_dict)
  File "/usr/local/bin/yt-dlp/yt_dlp/downloader/hls.py", line 77, in real_download
    urlh = self.ydl.urlopen(self._prepare_url(info_dict, man_url))
  File "/usr/local/bin/yt-dlp/yt_dlp/YoutubeDL.py", line 3863, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "/usr/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/usr/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/usr/lib/python3.9/urllib/request.py", line 561, in error
    return self._call_chain(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/usr/lib/python3.9/urllib/request.py", line 641, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 403: Forbidden
```

**Comment by bashonly** [comment]
`yt-dlp --update-to nightly` and try again

or as a workaround with the latest stable version, you can try adding this to your command:
`--add-header "Referer:https://twitter.com/"`
